pytest/tests/mocknet/sharded_bm.py

```diff
@@ -280,6 +283,84 @@ def start_nodes(args, enable_tx_generator=False):
     start_nodes_cmd(CommandContext(start_nodes_cmd_args))
 
 
+def handle_get_traces(args):
+    """Handle the get-traces command - fetch traces from the tracing server.
+    """
+    if args.forknet_details['tracing_server_external_ip'] is None:
+        logger.error("Error: Tracing server external IP is not set")
+        return
+
+    try:
+        if args.time is None:
+            cur_time = datetime.datetime.now(datetime.timezone.utc)
+        else:
+            cur_time = datetime.datetime.strptime(args.time,
+                                                  "%Y-%m-%d %H:%M:%S")
+            cur_time = cur_time.replace(tzinfo=datetime.timezone.utc)
+
+        cur_time_seconds = cur_time.timestamp()
+    except ValueError as e:
+        logger.error(
+            f"Invalid datetime format. Expected 'YYYY-MM-DD HH:MM:SS'. Error: {e}"
+        )
+        return
+
+    start_time = int((cur_time_seconds - args.lag - args.len) * 1000)
+    end_time = int((cur_time_seconds - args.lag) * 1000)
+
+    logger.info(
+        f"Start time: {datetime.datetime.fromtimestamp(start_time/1000, tz=datetime.timezone.utc).isoformat()}"
+    )
+    logger.info(
+        f"End time: {datetime.datetime.fromtimestamp(end_time/1000, tz=datetime.timezone.utc).isoformat()}"
+    )
+
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    trace_file = os.path.join(args.output_dir, f"trace_{start_time}.json")
+
+    response = requests.post(
+        f"http://{args.forknet_details['tracing_server_external_ip']}:8080/raw_trace",
+        headers={
+            'Content-Type': 'application/json',
+            'Accept-Encoding': 'gzip, deflate, br',
```

Comment:
Is this enough to make the trace be transferred as compressed file?

Reply:
Yeah. In fact `gzip, deflate` is enough.

Interestingly, `requests.post` handles (de)compression under the hood, so we can't easily see compressed file size. But I tried `response.raw.read_chunked` and it gave ~12 MB file with gibberish inside, so I assume it works.
Also, the logs show `Response encoding: gzip`.